src/ts/components/charge/Charge.ts

```diff
@@ -29,54 +26,19 @@ class Charge {
   handleInputAmount = (e: Event) => {
     e.preventDefault();
     const charge = (this.chargeInput as HTMLInputElement).valueAsNumber;
-    verifyCharge(charge);
-    this.convertRandomCharge(charge);
-  };
-
-  convertRandomCharge(charge: number) {
-    let totalAmount = 0;
-    this.totalCharge += charge;
-    while (totalAmount !== charge) {
-      const randomCoin = this.pickNumberInList();
-      totalAmount += randomCoin;
-      if (totalAmount > charge) {
-        totalAmount -= randomCoin;
-      } else if (totalAmount <= charge) {
-        this.coinsKindCount[randomCoin]++;
-      }
-    }
-
-    this.chargeView.showRandomChargeResult(this.coinsKindCount, this.totalCharge);
-    this.setCoinList();
-    this.setTotalCharge();
-  }
-
-  pickNumberInList(): number {
-    const coinList = [10, 50, 100, 500];
-    const randomNumber = Math.floor(Math.random() * coinList.length);
-    return coinList[randomNumber];
-  }
-
-  setCoinList() {
-    localStorage.setItem("COIN_LIST", JSON.stringify(this.coinsKindCount));
-  }
-
-  getCoinList() {
-    return JSON.parse(localStorage.getItem("COIN_LIST")) || { 10: 0, 50: 0, 100: 0, 500: 0 };
-  }
-
-  setTotalCharge() {
-    localStorage.setItem("TOTAL_CHARGE", JSON.stringify(this.totalCharge));
-  }
 
-  getTotalCharge() {
-    return JSON.parse(localStorage.getItem("TOTAL_CHARGE")) || 0;
-  }
+    validateCharge(charge);
+    showSnackbar(insertMoneyText(charge))
+    this.chargeInfo.convertRandomCharge(charge);
+    this.chargeView.showRandomChargeResult(this.chargeInfo.getCoinList(), this.chargeInfo.getTotalCharge());
```

Comment:
```suggestion
    this.chargeView.showRandomChargeResult(this.chargeInfo.coins, this.chargeInfo.totalCash);
```
getter를 사용해서 코인 정보와 금액을 가져오면 사용해보면 더 좋을거같아요~
coinList라는 네이밍도 애매한 면이 있다고 생각해서 coins로 바꿔봤고 totalCharge를 사용하려면 이미 충전된 금액이니 totalCharged가 더 나을거같아요. 저는 totalCash로 바꿔봤습니다 🙂

Reply:
getter를 써서 더 좋은 이유에 대해 알 수 있을까요?

Reply:
개인적으로 메서드를 사용하는것보다 직관적으로 코드를 읽어갈 수 있다고 생각해요~
chargeInfo에서 충분히 가지고 있을만한 정보기도 하구요 👀